### ISSUE TITLE
Context as a list of FHIR Resources

### ISSUE THREAD

**Issue by euvitudo** [cds-hooks/docs#36]
I would like to suggest that context be explicitly documented as a list of FHIR Resources, in order to more accurately interpret what is included in the list.

**Comment by isaacvetter** [comment]
Hey @euvitudo !

Does this (more recently created) issue #39, capture your suggestion?

Isaac

**Comment by euvitudo** [comment]
Hey @isaacvetter 
For the most part it does with the exception that it doesn't actually explicitly state that context shall be a list of FHIR Resources (read: it seems to avoid saying as much).  I'll be interested to see how the formal definition plays out.

**Comment by kpshek** [comment]
@euvitudo - I definitely see FHIR resources being included in `context`. However, I'm not sure we should limit `context` to be composed *only* of a list of FHIR resources.

Eg, this issue seems to propose `context` be:

```
"context": [
 {...}, // some FHIR resource
 {...} // another FHIR resource
]
```

However, I think there are scenarios where context can contain FHIR resources + additional data outside of FHIR. For example:

```
"context": {
 "allergyIntolerance": {...}, // FHIR allergy resource being documented
 "someDataOutsideOfFHIR": "foo"
}
```

Coming up with additional hooks will help us see clarity on this topic. However, allowing context to be an object as opposed to an array will allow for greater flexibility when defining hooks (especially hooks unique to a particular EHR).

**Comment by euvitudo** [comment]
@kpshek - The way I see it is that additional data can be represented by resources such as `Basic` (with extensions), `Binary`, `Media` and possibly others.

To wit: I don't see how a FHIR-sanctioned spec can use anything other than FHIR-sanctioned resources.  If FHIR needs to support a more generic resource than `Basic` (again, with extensions), then that should be added to the FHIR specification.

**Comment by kpshek** [comment]
@euvitudo - SMART is being published by FHIR as well and does not use FHIR exclusively in its representation of data (for instance, context data a SMART app receives). As mentioned in #37, we moved away from exclusively using FHIR for all representations.

**Comment by jmandel** [comment]
>To wit: I don't see how a FHIR-sanctioned spec can use anything other than FHIR-sanctioned resources. 

My take is that we want to be pragmatic and move quickly. Design choices we make here might well influence future direction for FHIR (e.g. There's actively consideration now about a mere condensed Parameters representation). But first we have to show that this stuff works.

**Comment by euvitudo** [comment]
From an implementer's point of view, I find that a specification that allows anything to be passed in makes my life more difficult, as my implementation needs to know how to handle the context or reasonably ignore it.  If the impl ignores the context data, what use is it during decision support?

A well-defined, well-known specification allows me to have a level of certainty that the data received by the service can be reasonably leveraged to perform the CDS operation.

**Comment by jmandel** [comment]
To be clear, we're not proposing allowing "anything" -- we're proposing that CDS Hooks should specify the content (ans leveraging FHIR resources wherever they fit).

**Comment by euvitudo** [comment]
Fair enough.  I would be helpful to have @kpshek define what he means by:
> FHIR resources + additional data outside of FHIR

**Comment by kpshek** [comment]
@euvitudo - By "FHIR resources + additional data outside of FHIR" what I meant was that I would be surprised to see `context` containing non-FHIR data, especially for hooks that are specific to a particular EHR. I don't concrete examples of this today because our definition and exploration of hooks has been limited thus far.

Non FHIR data aside, by allowing `context` to be a JSON object, it gives us more flexibility than an array.

**Comment by euvitudo** [comment]
@kpshek - that makes sense.  Thanks.

**Comment by euvitudo** [comment]
This will hopefully be addressed via #39.
